doc/cli_guidelines.md

```diff
@@ -0,0 +1,196 @@
+# D-Installer CLI Guidelines
+
+This document defines the syntax for the D-Installer CLI. For more CLI specific aspects like return values, flags, etc, please refer to [clig.dev](https://clig.dev/). Guidelines from clig.dev are agnostic about programming languages and tooling in general, and it can be perfectly used as reference for the D-Installer CLI.
+
+## CLI Syntax
+
+Note: The syntax presented here was previously discussed in this [document](https://gist.github.com/joseivanlopez/808c2be0cf668b4b457fc5d9ec20dc73).
+
+The installation settings are represented by a YAML structure, and the CLI is defined as a set of generic sub-commands and verbs that allow to edit any YAML value in a standard way.
+
+The CLI offers a `config` sub-command for editing the YAML config. The `config` sub-command has verbs for the following actions:
+
+* To load a YAML config file with the values for the installation.
+* To edit any value of the config without loading a new complete file again.
+* To show the current config for the installation.
+* To check the current config.
+
+Moreover, the CLI also offers sub-commands for these actions:
+
+* To ask for the possible values that can be used for some setting (e.g., list of available products).
+* To start and abort the installation.
+* To see the installation status.
+* To answers questions.
```

Comment:
I would probably add here also subcommand for validation. So you can ask if current config is valid and if not, to show details like wrong software selection or users issue.

Reply:
There is already a command for that in the list of commands: `dinstaller config check`.